### ISSUE TITLE
Slots are unavailable even though they are enabled in settings for event

### ISSUE THREAD

**Issue by erarta** [calcom/cal.com#8365]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

A summary of the issue. This needs to be a clear detailed-rich summary.

### Steps to Reproduce

1. Go to calendar https://cal.com/doctor-dubskaya/60min
2. You see only 21 april available for booking, but a lot more slots are enabled in settings but do not show up
3.  

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

### Technical details

- Browser version, screen recording, console logs, network requests: You can make a recording with [Bird Eats Bug](https://birdeatsbug.com/).
- Node.js version
- Anything else that you think could be an issue.
![Monosnap Working Hours | Доступность | Cal com 2023-04-19 10-28-46](https://user-images.githubusercontent.com/118567358/233001636-04778a5b-e2c2-4bab-9c92-2026fa0f4bbe.png)

![Monosnap Прием врача, 1 час, 4000р | Doctor Dubskaya | Cal com 2023-04-19 10-28-09](https://user-images.githubusercontent.com/118567358/233001655-7fe48621-a169-4493-942c-ffb947b33782.png)

**Comment by erarta** [comment]
Please tell me why these available slots do not show up. It is real emergency. Need to resolve as soon as possible.

**Comment by Airback** [comment]
I am suffering from the same bug. I was not able to resolve it by adjusting the settings of the event type.

**Comment by PeerRich** [comment]
we are looking into this

**Comment by vaibhav-kona** [comment]
I thought I too had this issue but in my case, I didn't save it actually. There is a save button on the top right of the page. I didn't save it and was expecting that the schedule got updated as I clicked on the "Update override" button. 


<img width="1229" alt="2023-04-30_10-07" src="https://user-images.githubusercontent.com/5622964/235335796-e3a181c4-5ae0-4970-b195-0bd870c57472.png">
<img width="1513" alt="2023-04-30_10-07_1" src="https://user-images.githubusercontent.com/5622964/235335797-68b087d6-5bbb-429d-8b10-b70aae66075a.png">

It could be easier if some warning shows on top of the "Date Overrides" section that something is updated and need to be saved to persist it. Also, the changes are getting lost when going to another tab. An alert can be shown to avoid going to other tabs/pages if there are any pending changes to be saved. @PeerRich Let me know if I should raise this as a separate feature suggestion if it's not in the plan for UI changes already.
